numpy/core/include/numpy/npy_common.h

```diff
@@ -10,13 +10,23 @@
 #include <npy_config.h>
 #endif
 
+// compile time environment variables
+#ifndef NPY_RELAXED_STRIDES_CHECKING
+    #define NPY_RELAXED_STRIDES_CHECKING 0
+#endif
+#ifndef NPY_RELAXED_STRIDES_DEBUG
+    #define NPY_RELAXED_STRIDES_DEBUG 0
+#endif
```

Comment:
I added `NPY_USE_NEW_CASTINGIMPL` in the meantime. We could probably also include it in `numpy/core/setup.py` so that it is all in one place (although it doesn't matter, easy enough to figure out if CI fails).